ksql-cli/src/test/java/io/confluent/ksql/cli/CliTest.java

```diff
@@ -792,9 +798,142 @@ public void shouldPrintErrorIfCantFindFunction() throws Exception {
+    // Given:
+    final String statementText = "create stream foo;";
+    final KsqlRestClient mockRestClient = givenMockRestClient();
+    final CommandStatusEntity stubEntity = stubCommandStatusEntityWithSeqNum(12L);
+    when(mockRestClient.makeKsqlRequest(anyString(), anyLong()))
+        .thenReturn(RestResponse.successful(new KsqlEntityList(
+            Collections.singletonList(stubEntity))));
+
+    // When:
+    localCli.handleLine(statementText);
+
+    final String secondStatement = "list streams;";
+    localCli.handleLine(secondStatement);
+
+    // Then:
+    verify(mockRestClient).makeKsqlRequest(secondStatement, 12L);
+  }
+
+  @Test
+  public void shouldUpdateCommandSequenceNumberOnMultipleCommandStatusEntities() throws Exception {
+    // Given:
+    final String statementText = "create stream foo;";
+    final KsqlRestClient mockRestClient = givenMockRestClient();
+    final CommandStatusEntity firstEntity = stubCommandStatusEntityWithSeqNum(12L);
+    final CommandStatusEntity secondEntity = stubCommandStatusEntityWithSeqNum(14L);
+    when(mockRestClient.makeKsqlRequest(anyString(), anyLong()))
+        .thenReturn(RestResponse.successful(new KsqlEntityList(
+            ImmutableList.of(firstEntity, secondEntity))));
+
+    // When:
+    localCli.handleLine(statementText);
+
+    final String secondStatement = "list streams;";
+    localCli.handleLine(secondStatement);
+
+    // Then:
+    verify(mockRestClient).makeKsqlRequest(secondStatement, 14L);
+  }
+
+  @Test
+  public void shouldNotUpdateCommandSequenceNumberIfNoCommandStatusEntities() throws Exception {
+    // Given:
+    final String statementText = "create stream foo;";
+    final KsqlRestClient mockRestClient = givenMockRestClient();
+    when(mockRestClient.makeKsqlRequest(anyString(), anyLong()))
+        .thenReturn(RestResponse.successful(new KsqlEntityList()));
+
+    // When:
+    localCli.handleLine(statementText);
+
+    final String secondStatement = "list streams;";
+    localCli.handleLine(secondStatement);
+
+    // Then:
+    verify(mockRestClient).makeKsqlRequest(secondStatement, -1L);
+  }
+
   private void givenRunInteractivelyWillExit() {
-    EasyMock.expect(lineSupplier.get()).andReturn("eXiT");
-    EasyMock.replay(lineSupplier);
+    when(lineSupplier.get()).thenReturn("eXiT");
+  }
+
+  private KsqlRestClient givenMockRestClient() throws Exception {
+    final KsqlRestClient mockRestClient = mock(KsqlRestClient.class);
+
+    when(mockRestClient.getServerInfo()).thenReturn(
+        RestResponse.of(new ServerInfo("1.x", "testClusterId", "testServiceId")));
+    when(mockRestClient.getServerAddress()).thenReturn(new URI("http://someserver:8008"));
+
+    localCli = new Cli(
+        STREAMED_QUERY_ROW_LIMIT, STREAMED_QUERY_TIMEOUT_MS, mockRestClient, console);
+
+    return mockRestClient;
+  }
+
+  private CommandStatusEntity stubCommandStatusEntityWithSeqNum(final long seqNum) {
+    return new CommandStatusEntity(
```

Comment:
I wanted this to return a mock and initially tried
```
  private CommandStatusEntity mockCommandStatusEntityWithSeqNum(final long seqNum) {
    final CommandStatusEntity mockEntity = mock(CommandStatusEntity.class);
    when(mockEntity.getCommandSequenceNumber()).thenReturn(seqNum);
    when(mockEntity.getCommandStatus()).thenReturn(mock(CommandStatus.class));
    return mockEntity;
  }
```
but the newly added tests were failing with 
```
java.lang.RuntimeException: Unexpected KsqlEntity class: 'io.confluent.ksql.rest.entity.CommandStatusEntity$MockitoMock$564394681'

	at io.confluent.ksql.cli.console.Console.printAsTable(Console.java:350)
	at io.confluent.ksql.cli.console.Console.printKsqlEntityList(Console.java:294)
	at io.confluent.ksql.cli.Cli.printKsqlResponse(Cli.java:328)
	at io.confluent.ksql.cli.Cli.handleStatements(Cli.java:248)
	at io.confluent.ksql.cli.Cli.handleLine(Cli.java:181)
	at io.confluent.ksql.cli.CliTest.shouldUpdateCommandSequenceNumber(CliTest.java:865)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```
which confuses me since `Console.java` certainly has a handler for the KsqlEntity `CommandStatusEntity`. Perhaps the class of the mock is subtly different from the class of a regular `CommandStatusEntity`?

Reply:
What does `.getClass` on the mock return?

I think you'll find it's a different type, i.e. `io.confluent.ksql.rest.entity.CommandStatusEntity$MockitoMock$564394681`

So... rather than using a mock, can you just create a `CommandStatusEntity` in the right state?